### ISSUE TITLE
Validators don't get called when the $viewValue is cleared (angular 1.2)

### ISSUE THREAD

**Issue by koenweyn** [formly-js/angular-formly#236]
When using custom validators, the validator function is not called when a user clears the input field.
This means that when the validator triggered a validation error before, the error will remain.
I created a sample exhibiting this behaviour here: http://jsbin.com/ducuqu/edit?html,css,js,output

The bug is caused by this line in utils.js:
```
return expression(viewValue || modelValue, modelValue, scope);
```
When viewValue becomes '', the validator only sees the modelValue (twice).

I'm guessing that changing that line would have huge impact, any suggestions on how to solve this?

**Comment by kentcdodds** [comment]
You are correct, this would have a huge impact... I'm about to cut a 5.2.0 release. I'll see if I can get this fixed as part of that.

**Comment by koenweyn** [comment]
The only thing I can come up with is creating a second formlyEval and using that on the validation triggers. If that's something you would do too, I can create a PR to fix this.

**Comment by kentcdodds** [comment]
You mean, a second one that doesn't do the `|| modelValue`? I'd have to give that some thought... Let me play around with that, this may be worthy of a breaking version change...

**Comment by ValentinH** [comment]
@kentcdodds What is the reason behind the ```|| modelValue``` ?

**Comment by kentcdodds** [comment]
I thought there was, but now I can't remember why I had it. I'll give it a little thought and testing, then I may release a 6.0.0...

**Comment by ValentinH** [comment]
Because if it was to prevent an undefined value, a ```angular.isDefined(viewValue) ? viewValue : modelValue``` would be safer...

**Comment by koenweyn** [comment]
And that would solve this bug.

**Comment by kentcdodds** [comment]
If I remember correctly, it was simply intended to be a convenience, but
now I realize that it's more of a problem than it is helpful. I'll double
check, and likely push a breaking change version to fix that.

- Kent C. Dodds

(Sent from my mobile device, please forgive typos or brevity)

**Comment by kentcdodds** [comment]
Ok, this and several other things (see the changelog) have been changed in angular-formly 6.0.0-beta.2. If you could please try this out, that would be great. Thanks!

**Comment by koenweyn** [comment]
Confirmed, fixes my problem.
But maybe it would be wise to change the 'else' part of `formlyEval` as well
```
  } else {
    return scope.$eval(expression, {
      $viewValue: viewValue || modelValue, //change this to $viewValue: viewValue too
      $modelValue: modelValue
    });
  }
```

**Comment by kentcdodds** [comment]
Thanks! Missed that. Glad I only did this in a beta release :-)

**Comment by kentcdodds** [comment]
It's been updated in 6.0.0-beta.4. Anything else you'd like to see changed in a major version change?

**Comment by koenweyn** [comment]
Nope, looks perfect now.

**Comment by kentcdodds** [comment]
Alrighty, officially released in 6.0.0 :-)
